src/main/java/com/j256/ormlite/db/Db2DatabaseType.java

```diff
@@ -74,4 +77,22 @@ public boolean isOffsetSqlSupported() {
 		// there is no easy way to do this in this database type
 		return false;
 	}
+
+	@Override
+	public FieldConverter getFieldConverter(DataPersister dataType, FieldType fieldType) {
+		// we are only overriding certain types
+		switch (dataType.getSqlType()) {
+			case LOCAL_DATE: // db2 doesn't support JDBC 4.2
+				return DataType.LOCAL_DATE_SQL.getDataPersister();
+			case LOCAL_TIME:
+				return DataType.LOCAL_TIME_SQL.getDataPersister();
+			case LOCAL_DATE_TIME:
+				return DataType.LOCAL_DATE_TIME_SQL.getDataPersister();
+			case OFFSET_TIME: // db2 doesn't seem to support TIME/STAMP WITH TIME ZONE
+			case OFFSET_DATE_TIME:
+				return null;
```

Comment:
This pattern is going to cause problems. Returning null leads to hard-to-debug NullPointerExceptions within ORMLite.

We have two options here:

1. Throw a friendly exception. It'll be a bit messy to take this into account in the tests though.

2. Make a best effort at converting. We should be able to easily emulate the behaviour of PostgreSQL and H2. They, at database level, just convert the timestamp to UTC and store it as such. On converting back it uses the system timezone. The timezone is never actually stored, the WITH TIMEZONE` didn't actually mean much in the first place. We could do the same here using a `FieldConverter` to `LocalTime`/`LocalDateTime`. We want to take `Instant` into account too (which doesn't have its own SQL type because it rightly doesn't need one) so perhaps a `FieldConverter` wrapper around the data persister would work better than just creating a new `DataPersister`. See #40 for an approach I took in another situation. I think this approach will work best. We won't be doing any worse than PostgreSQL and H2's efforts.

Reply:
Sure, no problem. I’ll test everything again when you manage to take a look.